scripts/deploy-demo.js

```diff
@@ -0,0 +1,67 @@
+const assert = require('node:assert/strict');
+const path = require('node:path');
+
+const DEMO_NAME = process.argv[2];
+const PANTHER_DEMO_HOST = process.env.PANTHER_DEMO_HOST;
+const PANTHER_DEMO_TOKEN = process.env.PANTHER_DEMO_TOKEN;
+const DS_PWD = process.env.DS_PWD;
+const USER_GROUP = 'public-proxy-user-group';
+
+assert(DEMO_NAME, 'Missing demo name CLI argument');
+assert(PANTHER_DEMO_HOST, 'PANTHER_DEMO_HOST env variable is required.');
+assert(PANTHER_DEMO_TOKEN, 'PANTHER_DEMO_TOKEN env variable is required.');
+
+const { upsertEntity, upsertLayout } = require('./network')(PANTHER_DEMO_HOST, PANTHER_DEMO_TOKEN);
+
+const main = async () => {
+    const demoPath = path.resolve(__dirname, '..', DEMO_NAME);
+    const workspacesManifestPath = path.resolve(demoPath, 'workspaces', 'meta.json');
+    const { workspaces } = require(workspacesManifestPath);
+
+    await deployDataSource(
+        require(path.resolve(demoPath, 'dataSource', 'dataSource.json')),
+        require(path.resolve(demoPath, 'dataSource', 'pdm.json'))
+    );
+
+    await Promise.all(workspaces.map(workspaceId => deployWorkspace(
+        require(path.resolve(demoPath, 'workspaces', workspaceId, 'workspace.json')),
+        require(path.resolve(demoPath, 'workspaces', workspaceId, 'ldm.json')),
+        require(path.resolve(demoPath, 'workspaces', workspaceId, 'workspaceAnalytics.json'))
+    )));
+};
+
+const deployDataSource = async (dataSource, pdm) => {
+    const id = dataSource.data.id;
+
+    // Patch data source credentials
+    if (DS_PWD) {
+        dataSource.data.attributes.password = DS_PWD;
+    }
+
+    await upsertEntity('/dataSources', id, dataSource);
+    await upsertLayout(`/dataSources/${id}/physicalModel`, pdm);
+};
+
+const deployWorkspace = async (workspace, ldm, analytics) => {
+    const id = workspace.data.id;
+
+    await upsertEntity('/workspaces', id, workspace);
+    await upsertLayout(`/workspaces/${id}/permissions`, {
+        permissions: [{
```

Comment:
@danh-ung , I've added the needed permission for the public proxy user in here instead of creating a new JSON file under workspace. The reason being that this permission is specific to the proxy. I'm not sure this is the best solution, though. What do you think?

Reply:
yes, make sense